### PR TITLE
fix(KFLUXBUGS-1218): add support for pipelineRun timeouts

### PR DESCRIPTION
**Pull request by dirgim**
* Integration pipelineRuns can now apply default timeouts for pipeline, task and finally based on environment variables

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [x] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Comment by dirgim** [comment]
Related changes at https://github.com/redhat-appstudio/infra-deployments/pull/3655

File: tekton/integration_pipeline.go [hunk 1]
```diff
@@ -246,3 +247,32 @@ func (r *IntegrationPipelineRun) WithEnvironmentAndDeploymentTarget(dt *applicat
 
 	return r
 }
+
+// WithDefaultIntegrationTimeouts fetches the default Integration timeouts from the environment variables and adds them
+// to the integration PipelineRun.
+func (r *IntegrationPipelineRun) WithDefaultIntegrationTimeouts() *IntegrationPipelineRun {
+	pipelineTimeoutStr := os.Getenv("PIPELINE_TIMEOUT")
+	taskTimeoutStr := os.Getenv("TASKS_TIMEOUT")
+	finallyTimeoutStr := os.Getenv("FINALLY_TIMEOUT")
```
Comment (MartinBasti) [inline tekton/integration_pipeline.go:256]: Would be nice to have default values when ENVVARS are undefined.

Comment (MartinBasti) [reply tekton/integration_pipeline.go:256]: nevermind :D

File: tekton/integration_pipeline_test.go [hunk 2]
```diff
@@ -330,6 +332,41 @@ var _ = Describe("Integration pipeline", func() {
...
+			os.Setenv("FINALLY_TIMEOUT", "2h")
+			expectedDuration, _ := time.ParseDuration("2h")
+			newIntegrationPipelineRun.WithDefaultIntegrationTimeouts()
+
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Pipeline.Duration).To(Equal(expectedDuration))
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Tasks.Duration).To(Equal(expectedDuration))
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Finally.Duration).To(Equal(expectedDuration))
+
+			// Reset the environment variables
+			os.Setenv("PIPELINE_TIMEOUT", "")
```
Comment (MartinBasti) [inline tekton/integration_pipeline_test.go:365]: This should be reworked to proper Before/AfterEach, or `defer` so we don't have envvars messing around when a Expect fails in this test case

File: tekton/integration_pipeline.go [hunk 3]
```diff
@@ -19,16 +19,17 @@ package tekton
...
 	"github.com/redhat-appstudio/integration-service/api/v1beta2"
 	"github.com/redhat-appstudio/operator-toolkit/metadata"
 	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"reflect"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+	"time"
```
Comment (MartinBasti) [inline tekton/integration_pipeline.go:32]: IDE fight, I like the original way more

Comment (dirgim) [reply tekton/integration_pipeline.go]: Missed this one, reverted it back to how it was.

File: tekton/integration_pipeline_test.go (hunk 2 above)
Comment (dirgim) [reply tekton/integration_pipeline_test.go]: Good catch, moved the envvars setting to Before and AfterEach.

File: tekton/integration_pipeline_test.go [hunk 4]
```diff
@@ -330,6 +339,43 @@ var _ = Describe("Integration pipeline", func() {
...
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Pipeline).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Tasks).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Finally).To(BeNil())
+
+			expectedLogEntry := "failed to parse default PIPELINE_TIMEOUT"
+			Expect(buf.String()).Should(ContainSubstring(expectedLogEntry))
+			expectedLogEntry = "failed to parse default TASKS_TIMEOUT"
+			Expect(buf.String()).Should(ContainSubstring(expectedLogEntry))
+			expectedLogEntry = "failed to parse default FINALLY_TIMEOUT"
+			Expect(buf.String()).Should(ContainSubstring(expectedLogEntry))
```
[the remark this answers is not in the document]
Comment (dirgim) [reply tekton/integration_pipeline_test.go]: Ack, updated

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by MartinBasti — APPROVED** [review]

**Review by Josh-Everett — APPROVED** [review]

**Pull request merged**